### ISSUE TITLE
Fail to load api page after `azd up` for `todo-nodejs-mongo-aks`

### ISSUE THREAD

**Issue by Menghua1** [Azure/azure-dev#4459]
**Describe the issue:**
After running `azd up`, click the api link in the log, the page fails to load, and the page is as follows:
![Image](https://github.com/user-attachments/assets/22237836-6b9f-46be-978f-e993a26de2f0)

**Repro Steps:**
1. Run `azd init -t todo-nodejs-mongo-aks -b staging`.
2. Run `azd auth login`.
3. Run `azd up`.
4. Click the api link in the log.

**Environment:**
- Template: `todo-nodejs-mongo-aks`.
- Branch: `staging`
- OS: Windows.
- Azd version:  azd version 1.10.3-pr.4243081 (commit 2c05c152915bdfd82e20d3350f249b22accd3deb) and azd version 1.11.0-beta.1-daily.4244391 (commit 8fc9d80640aa6f45f1a7d7631087c4a5f54e9e63).

**Expected behavior:**
The api page can be loaded successfully.

@jongio for notification.

**Comment by vhvb1989** [comment]
This is most likely a regression in the template's infrastructure.  @wbreza could you take a look?

**Comment by wbreza** [comment]
@Menghua1 - are you deploying this to a Microsoft tenant subscription?  If so you won't be able to connect to this outside of internal network and requires connecting via a secure DevBox or similar.

**Comment by Menghua1** [comment]
@wbreza Yes, but the front-end page can be loaded normally. According to the investigation, the Request URLs of the requests `swagger-ui.css`, `swagger-ui-init.js`, `swagger-ui-bundle.js`, `swagger-ui-standalone-preset.js` seem to be incorrect. They should be the API link plus a suffix, not the front-end link. The response of these requests seems to be the files of the front-end page, which is also wrong.

- Web page:
![Image](https://github.com/user-attachments/assets/ae3ba5b3-c042-469d-87c5-55375f2bc110)

- Api page:
![Image](https://github.com/user-attachments/assets/e5a34e9e-b73c-4fc6-b3f2-ff5652b9b994)
![Image](https://github.com/user-attachments/assets/6d15a3d5-60df-4942-b56a-9553f74ee506)

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: `staging`.
- OS: `DevContainer`.
- Azd version: azd version 1.11.0-beta.1-daily.4247716 (commit https://github.com/Azure/azure-dev/commit/f4695c408eab9ea3cdedd1c3cb39aa01ad90123e).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Cloud: `AzureCloud`, `AzureChinaCloud`.
- Azd version: azd version 1.12.0-beta.1-daily.4359062 (commit https://github.com/Azure/azure-dev/commit/6cd82ae9ff02e156a0056e994215191469442ab3).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Cloud: `AzureCloud`, `AzureChinaCloud`.
- Azd version: azd version 1.12.0-beta.1-daily.4442669 (commit 9b60cbc2a703200eb04d325988dc359886b2dfcb).

**Comment by NanaXiong00** [comment]
@vhvb1989 and @wbreza, The current issue includes two problems:
1. API Service Exposure Path Issue: The application's API service is mapped to "/api", but the API page request path not include "/api", causing the API page to load as blank.
    - Expected API Request URL: `http://20.161.208.156/api/swagger-ui-init.js`
    - Actual API Request URL: `http://20.161.208.156/swagger-ui-init.js`
The error is as follows：
![Image](https://github.com/user-attachments/assets/d1051056-8da9-4bdd-8935-5947e1d7c341)

2. Swagger Request URL Issue: The Swagger-generated API request URLs are missing the "/api" prefix, preventing requests from correctly matching the backend service, which leads to API call failures.
    - Expected Swagger Request URL: `http://20.161.208.156/api/lists?top=20&skip=0`
    - Actual Swagger Request URL: `http://20.161.208.156/lists?top=20&skip=0`
The error is as follows：
![Image](https://github.com/user-attachments/assets/c91b9c6f-916d-43fe-a7e9-468bfe35ea15)

We will fix it using the following approach. If you think this approach is OK, we will prepare a PR for it.
- Modify the path in the Kubernetes Ingress configuration in [azure.yaml](https://github.com/Azure-Samples/todo-nodejs-mongo-aks/blob/main/azure.yaml) to "api/". 
![Image](https://github.com/user-attachments/assets/3588fab6-5719-4da8-bdfc-94e3fda41522)

- Add the "/api" prefix to the paths of each operation in [openapi.yaml](https://github.com/Azure-Samples/todo-nodejs-mongo-aks/blob/main/src/api/openapi.yaml).
![Image](https://github.com/user-attachments/assets/a59f2282-fb3c-4978-b75d-b8906f4b1ca2)

CC: @rajeshkamal5050

**Comment by vhvb1989** [comment]
Looks like a good fix to me @NanaXiong00 .  I would just wait for @wbreza confirmation. I am not 100% sure if after setting the `ingress` to `api/` you would still need to update all paths with `/api/`  or if that would ended up duplicating `api/api/` in the calls

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Cloud: `AzureCloud`, `AzureChinaCloud`.
- Azd version: azd version 1.12.0-beta.1-daily.4489231 (commit a6eb17988fc047579a8edaaa86f302f5f0ab4bb8).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.14.0-beta.1-daily.4694364 (commit 0340065907dee7ca71cd68ed132033cabb38cbce).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: Codespaces.
- Azd version: azd version 1.16.1-daily.4919861 (commit f2582c2c9a504b3958d8b4d3958a6c1ba4a1ec81).

**Comment by Menghua1** [comment]
> Looks like a good fix to me [@NanaXiong00](https://github.com/NanaXiong00) . I would just wait for [@wbreza](https://github.com/wbreza) confirmation. I am not 100% sure if after setting the `ingress` to `api/` you would still need to update all paths with `/api/` or if that would ended up duplicating `api/api/` in the calls

Just confirmed that even if the ingress path is set to `/api/`, we still need to add the `/api` prefix to all paths in `openapi.yaml`. This will not result in duplicate `/api/api/` paths, as shown below:

![Image](https://github.com/user-attachments/assets/fa390b5d-c07e-4a0f-8f93-c16712493e48)

@wbreza Could you please help confirm if this approach looks good to you? If so, we’ll proceed with the PR.

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: Codespaces.
- Azd version: azd version 1.17.2-daily.5010470 (commit f60fb010feb98e1bbd4dabeb7e3decb2b9fb096a).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.19.0-beta.1-daily.5116963 (commit c9f3849e1f45c071c1eb2ead613fe111fc883f73).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Template: `todo-nodejs-mongo-aks`.
- Branch: staging.
- OS: DevContainer.
- Azd version: azd version 1.19.0-beta.1-daily.5273554 (commit a9a799fcecb24305bdf57298453574e3b0ac9622).

**Comment by vhvb1989** [comment]
please move this issue to https://github.com/Azure-Samples/todo-nodejs-mongo-aks

**Comment by Menghua1** [comment]
@vhvb1989 It has been moved to the repo `todo-nodejs-mongo-aks`: https://github.com/Azure-Samples/todo-nodejs-mongo-aks/issues/17, so I am closing this issue.
